### ISSUE TITLE
net::ERR_HTTP2_FRAME_SIZE_ERROR on statistics console if acessed with H2 since 3.x

### ISSUE THREAD

**Issue by bruno-r-silva-cuf-pt** [haproxy/haproxy#2757]
### Detailed Description of the Problem

if the statistics console is acessed with h2, the page is cut to the header and just a couple of lines, and the browsers with will report net::ERR_HTTP2_FRAME_SIZE_ERROR (tested with both chrome and firefox). it still returns http 200.

forcing the backend with alpn http1.1 gets the full page back.

it seems to have started in the 3.x versions. 2.9.9 works fine in h2 and even quic.

### Expected Behavior

get the full statistics page, even via h2 or even quic

### Steps to Reproduce the Behavior

enable a backend with h2 to access the stats page, and access it with chrome or firefox

### Do you have an idea how to solve the issue?

quick fix is to force http 1.1. but my production setups are usually more complicated than this: I have a global frontend on 443 with h2 and quic, and then route to this backend if the hostname has a particular format.

### What is your configuration?

```haproxy
global
	server-state-file	/var/lib/haproxy/server.state

	log		127.0.0.1	local0 info

	chroot		/var/lib/haproxy/chroot
	maxconn		8000
	user		haproxy
	group		haproxy
	#debug

	ssl-default-bind-ciphers ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256
	ssl-default-bind-options no-sslv3 no-tlsv10 no-tlsv11 no-tls-tickets

	tune.ssl.default-dh-param	2048


	# turn on stats unix socket
	stats socket	/var/lib/haproxy/stats level admin expose-fd listeners

	tune.maxrewrite	16384
	tune.bufsize	32768

defaults
	load-server-state-from-file	global
	option	dontlognull
	option	redispatch
	option	log-separate-errors
	maxconn	58000

	retries 3
	backlog 10000
	timeout client 30s
	timeout connect 10s
	timeout server 35s
	timeout tunnel 10m
	timeout http-keep-alive 10s
	timeout http-request 15s
	timeout queue 30s
	timeout check 10s
	timeout tarpit 2m

	default-server weight 100 inter 3s rise 2 fall 3 inter 5s fastinter 2s slowstart 30s alpn h2 ssl verify none check check-ssl check-alpn h2 agent-check agent-port 7777 agent-inter 7s

	errorfile 500 /etc/haproxy/errors-custom/500.http
	errorfile 503 /etc/haproxy/errors-custom/503.http


listen haproxy
        mode http
        no log

        bind            *:443 ssl alpn h2,http1.1 crt /etc/haproxy/certs # does not work
        bind            *:443 ssl alpn http1.1 crt /etc/haproxy/certs # works

        acl is-basic-auth-user http_auth(stats-auth-list)

        acl is-admin-user  http_auth_group(stats-auth-list) admin-group
        acl is-viewer-user  http_auth_group(stats-auth-list) viewer-group

        http-request auth realm "HAProxy Stats" if !is-basic-auth-user

        stats   enable
        stats   refresh 30s
        stats   uri /haproxy?stats
        stats   show-node
        stats   show-legends
        stats   show-modules
        stats   admin if is-admin-user
```
```


### Output of `haproxy -vv`

```plain
HAProxy version 3.0.5-8e879a5 2024/09/19 - https://haproxy.org/
Status: long-term supported branch - will stop receiving fixes around Q2 2029.
Known bugs: http://www.haproxy.org/bugs/bugs-3.0.5.html
Running on: Linux 5.15.85-1-pve #1 SMP PVE 5.15.85-1 (2023-02-01T00:00Z) x86_64
Build options :
  TARGET  = linux-glibc
  CC      = cc
  CFLAGS  = -O2 -g -fwrapv
  OPTIONS = USE_OPENSSL=1 USE_LUA=1 USE_ZLIB=1 USE_TFO=1 USE_NS=1 USE_SYSTEMD=1 USE_QUIC=1 USE_PROMEX=1 USE_STATIC_PCRE2=1 USE_PCRE2_JIT=1
  DEBUG   =

Feature list : -51DEGREES +ACCEPT4 +BACKTRACE -CLOSEFROM +CPU_AFFINITY +CRYPT_H -DEVICEATLAS +DL -ENGINE +EPOLL -EVPORTS +GETADDRINFO -KQUEUE -LIBATOMIC +LIBCRYPT +LINUX_CAP +LINUX_SPLICE +LINUX_TPROXY +LUA +MATH -MEMORY_PROFILING +NETFILTER +NS -OBSOLETE_LINKER +OPENSSL -OPENSSL_AWSLC -OPENSSL_WOLFSSL -OT -PCRE +PCRE2 +PCRE2_JIT -PCRE_JIT +POLL +PRCTL -PROCCTL +PROMEX -PTHREAD_EMULATION +QUIC -QUIC_OPENSSL_COMPAT +RT +SHM_OPEN -SLZ +SSL -STATIC_PCRE +STATIC_PCRE2 +SYSTEMD +TFO +THREAD +THREAD_DUMP +TPROXY -WURFL +ZLIB

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_TGROUPS=16, MAX_THREADS=256, default=28).
Built with OpenSSL version : OpenSSL 1.1.1w+quic  11 Sep 2023
Running on OpenSSL version : OpenSSL 1.1.1w+quic  11 Sep 2023
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : TLSv1.0 TLSv1.1 TLSv1.2 TLSv1.3
Built with Lua version : Lua 5.4.7
Built with the Prometheus exporter as a service
Built with network namespace support.
Built with zlib version : 1.2.11
Running on zlib version : 1.2.11
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built with PCRE2 version : 10.44 2024-06-07
PCRE2 library supports JIT : yes
Encrypted password support via crypt(3): yes
Built with gcc compiler version 8.5.0 20210514 (Red Hat 8.5.0-4)

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
       quic : mode=HTTP  side=FE     mux=QUIC  flags=HTX|NO_UPG|FRAMED
         h2 : mode=HTTP  side=FE|BE  mux=H2    flags=HTX|HOL_RISK|NO_UPG
  <default> : mode=HTTP  side=FE|BE  mux=H1    flags=HTX
         h1 : mode=HTTP  side=FE|BE  mux=H1    flags=HTX|NO_UPG
       fcgi : mode=HTTP  side=BE     mux=FCGI  flags=HTX|HOL_RISK|NO_UPG
  <default> : mode=TCP   side=FE|BE  mux=PASS  flags=
       none : mode=TCP   side=FE|BE  mux=PASS  flags=NO_UPG

Available services : prometheus-exporter
Available filters :
	[BWLIM] bwlim-in
	[BWLIM] bwlim-out
	[CACHE] cache
	[COMP] compression
	[FCGI] fcgi-app
	[SPOE] spoe
	[TRACE] trace

---
Linux haproxy8.v.some.domain 5.15.85-1-pve #1 SMP PVE 5.15.85-1 (2023-02-01T00:00Z) x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by bruno-r-silva-cuf-pt** [comment]
note: my config files are a lot bigger, with more backends and frontends.  I only sent the settings related with the statistics reports page, but any reasonable set of backends and frontends will trigger this.

**Comment by bruno-r-silva-cuf-pt** [comment]
this trick also circunvents it:


```
frontend f_web443
	bind		*:443 ssl alpn h2 allow-0rtt ecdhe secp384r1 crt /etc/haproxy/certs
	bind		quic4@:443 ssl alpn h3 crt /etc/haproxy/certs

	mode		http

        use_backend     haproxy if { path_beg /haproxy }
        use_backend     %[path,lower,map_beg(/etc/haproxy/backendsmap.cfg,b_badmap)]

backend haproxy
	mode http
	no log

	#redirect to haproxy_real
	server localhost	127.0.0.1:8181 alpn http1.1 no-ssl no-check

frontend haproxy_real
	mode http
	no log

	bind 127.0.0.1:8181	alpn http1.1

	acl is-basic-auth-user http_auth(stats-auth-list)

	acl is-admin-user  http_auth_group(stats-auth-list) admin-group
	acl is-viewer-user  http_auth_group(stats-auth-list) viewer-group

	http-request auth realm "HAProxy Stats" if !is-basic-auth-user

	stats	enable
	stats	refresh 30s
	stats	uri /haproxy?stats
	stats	show-node
	stats	show-legends
	stats	show-modules
	stats	admin if is-admin-user

```

**Comment by capflam** [comment]
I'm able to reproduce the issue with your buffer size (32k) and with the zero-copy data forwarding. Waiting for a proper fix, you can disable it by starting haproxy with `-dZ` option, or by adding ` tune.disable-zero-copy-forwarding` directive in your global section.

**Comment by bruno-r-silva-cuf-pt** [comment]
> I'm able to reproduce the issue with your buffer size (32k) and with the zero-copy data forwarding. Waiting for a proper fix, you can disable it by starting haproxy with `-dZ` option, or by adding ` tune.disable-zero-copy-forwarding` directive in your global section.

confirmed. 
tune.disable-zero-copy-forwarding "fixed" it, so did commenting out tune.bufsize (and also commenting disable zero-copy, leaving it on)

**Comment by capflam** [comment]
It should be fixed in 3.1. Thanks !
